docs/source/architecture.rst

```diff
@@ -0,0 +1,143 @@
+interface to the driver's unified test runner.
+
+The workload executor is invoked by Astrolabe once a test Atlas cluster is
+up and running. The workload executor:
+
+- Instantiates the driver's unified test runner.
+- Receives the workload specification, which is a unified test format test,
+  from Astrolabe, and passes it to the unified test runner.
+- Sets up a termination signal handler.
+- Invokes the unified test runner to execute the workload, which ordinarily
+  contains a ``loop`` operation.
+- When the termination signal is received from Astrolabe, instructs the
+  unified test runner to stop looping.
+- Collects the number of iterations and successful operations performed by
+  the test runner, the number of errorsr and failures, the events produced
+  by the driver while executing the workload, and writes all of this
+  to files.
+
+
+Unified Test Runners
+--------------------
+
+On the driver side, the test workload is executed by the driver's
+`unified test runner <https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst>`_.
+The workload definitions must be valid unified test format tests with
+some additional constraints to simplify the implementation of Astrolabe
+as well as workload executors.
+
+The unified test format specification has been extended to provide several
+special operations, most importantly the ``loop`` operation, and options
+needed for Drivers Atlas Testing.
+
+
+Test Scenario
+-------------
+
+The :doc:`test scenario <spec-test-format>` describes the test to be performed.
+It consists of the following components:
+
+- Initial Atlas cluster configuration. Used by Astrolabe when creating the
+  cluster.
+- Atlas maintenance operations to be performed. The operations are
+  parsed and executed by Astrolabe.
+- Driver workload to execute, which is a valid (and complete)
+  unified test format test.
+
+There are some additional requirements imposed by Drivers Atlas Testing on
+the test scenarios to simplify the implementation of both Astrolabe and
+the workload executors, namely:
+
+- There must be exactly one ``loop`` operation per test scenario.
+- The names of entities used for retrieving events and statistics are
+  explicitly defined and fixed.
+
+
+Results & Artifacts
+-------------------
+
+Each test run produces the following results and artifacts:
+
+- Success or failure indication. Any errors or unified test runner failures
+  encountered during the test run fails the test run.
+- Execution log. This is useful for debugging but, because driver operations
+  are performed in an infinite loop, is generally not useful for higher-level
+  analysis of the impact of Atlas maintenance on applications.
+- Server logs for all nodes in the cluster and FTDC data. Currently
+  Drivers Atlas Testing does not use these logs but they are stored for
+  troubleshooting or investigative purposes.
+- Operation statistics: number of loop iterations performed, number of
```

Comment:
How are the statistics distinct from the `Success or failure indication`? IIUC we derive the success or failure indiciation from the statistics so I recommend you keep this bullet point but delete the first one (i.e. delete `Success or failure indication`)